src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/util/GraphQLRequestParamUtils.java

```diff
@@ -79,28 +81,27 @@ public static boolean isGraphQLContentType(final String contentType) {
      * @throws RuntimeException if JSON serialization fails for some reason due to any runtime environment issues
      */
     public static String toPostBodyString(final GraphQLRequestParams params) {
-        final ObjectMapper mapper = new ObjectMapper();
-        final ObjectNode postBodyJson = mapper.createObjectNode();
-        postBodyJson.set(OPERATION_NAME_FIELD,
-                JsonNodeFactory.instance.textNode(StringUtils.trimToNull(params.getOperationName())));
-
-        if (StringUtils.isNotBlank(params.getVariables())) {
-            try {
-                final ObjectNode variablesJson = mapper.readValue(params.getVariables(), ObjectNode.class);
-                postBodyJson.set(VARIABLES_FIELD, variablesJson);
-            } catch (JsonProcessingException e) {
-                log.error("Ignoring the GraphQL query variables content due to the syntax error: {}",
-                        e.getLocalizedMessage());
+        final JsonFactory jsonFactory = new JsonFactory();
+        final StringWriter writer = new StringWriter(1024);
+
+        try (JsonGenerator gen = jsonFactory.createGenerator(writer)) {
+            gen.writeStartObject();
+
+            gen.writeStringField(OPERATION_NAME_FIELD, StringUtils.trimToNull(params.getOperationName()));
+
+            if (StringUtils.isNotBlank(params.getVariables())) {
+                gen.writeRaw(",\"" + VARIABLES_FIELD + "\":");
```

Comment:
Just wondering: does writeFieldName work here?

Reply:
I tried with that at first to find out it didn't work. When writing a raw data, fieldName should be written together.

Reply:
This works:

```java
                gen.writeFieldName(VARIABLES_FIELD);
                gen.writeRawValue(StringUtils.trim(params.getVariables()));
```